glean/src/core/events/index.ts

```diff
@@ -76,13 +77,19 @@ export class CoreEvent<
  */
 const CoreEvents: {
   afterPingCollection: CoreEvent<[PingPayload], Promise<JSONObject>>,
+  afterPingInfoCollection: CoreEvent<[PingInfo, ExperimentData?], Promise<JSONObject>>,
   [unused: string]: CoreEvent
 } = {
   // Event that is triggered immediatelly after a ping is collect and before it is recorded.
   //
   //  - Context: The `PingPayload` of the recently collected ping.
   //  - Result: The modified payload as a JSON object.
-  afterPingCollection: new CoreEvent<[PingPayload], Promise<JSONObject>>("afterPingCollection")
+  afterPingCollection: new CoreEvent<[PingPayload], Promise<JSONObject>>("afterPingCollection"),
+  // Event that is triggered immediatelly after a ping_info is collect and before it is recorded.
+  //
+  //  - Context: The `PingPayload` of the recently collected ping.
+  //  - Result: The modified payload as a JSON object.
+  afterPingInfoCollection: new CoreEvent<[PingInfo, ExperimentData?], Promise<JSONObject>>("afterPingInfoCollection")
```

Comment:
The context on this event should not know anything about the experiment data itself. We can change this to

```suggestion
  afterPingInfoCollection: new CoreEvent<[PingInfo], Promise<JSONObject>>("afterPingInfoCollection")
```

Then the plugin will take that ping_info and modify it as needed. Does it make sense?